concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+        FROM `physionet-data.eicu_crd.admissiondx`
+        GROUP BY patientunitstayid
+      )
+  
+    SELECT patientunitstayid, admit_text1, admit_text2, admit_text3
+    , CASE 
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|No%" THEN NULL
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|Yes%" 
+      AND ( (LOWER(admit_text1) LIKE "%surgery%" OR LOWER(admit_text2) LIKE "%surgery%" OR LOWER(admit_text3) LIKE "%surgery%")
+      OR  (LOWER(admit_text1) LIKE "%operative%" OR LOWER(admit_text2) LIKE "%operative%" OR LOWER(admit_text3) LIKE "%operative%") )
+      THEN 1
+    ELSE NULL
+    END AS electivesurgery2
+
+    , CASE 
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|No%" THEN NULL
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|Yes%" THEN 1
+    ELSE NULL
+    END AS adm_elective2
+
+    FROM tt
+    )
+    AS adm
+    ON adm.patientunitstayid = apache.patientunitstayid
+
+    -- 2: patient table
+    FULL OUTER JOIN (
+      SELECT patientunitstayid
+      , CASE
+      WHEN 
+        (FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC) LIKE "Emergency Department"
+      OR FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC) LIKE "Direct Admit"
+      OR FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC) LIKE "Chest Pain Center"
+      OR FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC) LIKE "Other Hospital"
+      OR FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC) LIKE "Observation")
+      THEN NULL
+      ELSE 1
+      END AS adm_elective1
+      FROM `physionet-data.eicu_crd.patient`
+      WHERE unitAdmitSource IS NOT NULL
+        AND patientunitstayid IS NOT NULL 
+      GROUP BY patientunitstayid, unitAdmitSource
+    )
+    AS pat
+    ON pat.patientunitstayid = apache.patientunitstayid
+
+-- Statement for 1: apachepredvar table
+  WHERE apache.patientunitstayid IS NOT NULL AND adm.patientunitstayid IS NOT NULL
+)
+
+, electivesurgery_OASIS AS (
+  SELECT patientunitstayid AS pid_adm
+  , CASE
+    WHEN electivesurgery1 IS NULL THEN 6
+    WHEN electivesurgery2 IS NULL THEN 6
+    WHEN adm_elective1 IS NULL THEN 6
+    WHEN adm_elective2 IS NULL THEN 6
+    ELSE 0
+    END AS electivesurgery_OASIS
+  FROM elective_surgery
+)
+
+-- Heart rate 
+, heartrate_OASIS AS (
+SELECT patientunitstayid AS pid_HR
+  , CASE
+    WHEN MIN(heartrate < 33) THEN 4
+    WHEN MAX(heartrate) BETWEEN 33 AND 88 THEN 0
+    WHEN MAX(heartrate) BETWEEN 89 AND 106 THEN 1
+    WHEN MAX(heartrate) BETWEEN 107 AND 125 THEN 3
+    WHEN MAX(heartrate >125) THEN 6
+    ELSE NULL
+    END AS heartrate_OASIS
+  FROM `physionet-data.eicu_crd_derived.pivoted_vital`
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND heartrate IS NOT NULL
+    AND patientunitstayid IS NOT NULL 
+  GROUP BY pid_HR, heartrate
```

Comment:
You want: one row per `pid_HR`, i.e. one row per patient stay. So only group by `pid_HR`. If you also group by `heartrate`, you'll get one row for every unique `pid_HR` / `heartrate` pair, which is *not* what you want, as you want to get the min or max heart rate.